http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/SamAuthProvider.scala

```diff
@@ -207,6 +208,23 @@ class SamAuthProvider[F[_]: OpenTelemetryMetrics](
   )(implicit sr: SamResource[R], ev: Ask[F, TraceId]): F[Unit] =
     samDao.deleteResource(samResource, creatorEmail, googleProject)
 
+  override def lookupOriginatingUserEmail[R](petOrUserInfo: UserInfo)(implicit ev: Ask[F, TraceId]): F[WorkbenchEmail] =
+    for {
+      traceId <- ev.ask
+      samUserInfo <- samDao
+        .getSamUserInfo(petOrUserInfo.accessToken.token)
+        .flatMap(
+          _.fold(
+            F.raiseError[SamUserInfo](
+              AuthProviderException(
+                traceId,
+                s"[SamAuthProvider.lookupOriginatingUserEmail] Subject info not found for ${petOrUserInfo.userEmail}",
+                StatusCodes.Unauthorized
+              )
+            )
+          )(s => F.pure(s))
```

Comment:
syntax wise, this might be a bit easier to read

```
      samUserInfoOpt <- samDao.getSamUserInfo(petOrUserInfo.accessToken.token)
      samUserInfo <- F.fromOption(
        samUserInfoOpt,
        AuthProviderException(
          traceId,
          s"[SamAuthProvider.lookupOriginatingUserEmail] Subject info not found for ${petOrUserInfo.userEmail}",
          StatusCodes.Unauthorized
        )
      )
```